lib/puppet/network/http/api/v3/indirected_routes.rb

```diff
@@ -88,6 +87,8 @@ def uri2indirection(http_method, uri, params)
       params[:environment] = configured_environment
     end
 
+    check_authorization(method, "#{self.class.url_prefix}/#{indirection_name}/#{key}", params)
+
```

Comment:
Previously we checked authorization fairly early on, but now we have work to do to lookup the environment object. If there is a problem we raise an exception, e.g. ["Could not find environment"](https://github.com/camlow325/puppet/blob/bug/master/PUP-3855-auth-conf-not-honoring-environment/lib/puppet/network/http/api/v3/indirected_routes.rb#L84) which would allow an unauthenticated user the ability to enumerate environment names. Should we always raise not authorized instead?